wordpress-mobile/WordPress-iOS
# Update blogging prompts to use the v3 version

Refs #20496

> **Warning**
This PR depends on a development version of WordPressKit: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/592

As titled, this updates the blogging prompts service to use the v3 version. Please note that the new endpoint version **will return a different set of prompts from v2** despite having the same date range. Additionally:
- Removed unused `title` and `content` properties from the `BloggingPrompt` Core Data model. These were intended to be a placeholder for the prompt answer draft but were ultimately unused.
- Updated the unit tests to reflect the v3 changes.
- For locally scheduled prompts, any pending prompt reminders should be rescheduled and updated on app launch. See: [`BloggingPromptCoordinator#updatePromptsIfNeeded`](https://github.com/wordpress-mobile/WordPress-iOS/blob/901e6bf5ec58e043ee02e4529a88f45e2b5f61c8/WordPress/Classes/ViewRelated/Blog/Blogging%20Prompts/BloggingPromptCoordinator.swift#L82) and [`PromptRemindersScheduler#processSchedule`](https://github.com/wordpress-mobile/WordPress-iOS/blob/901e6bf5ec58e043ee02e4529a88f45e2b5f61c8/WordPress/Classes/Utility/Blogging%20Prompts/PromptRemindersScheduler.swift#L148)

### Todo

- [ ] Remove the blogging prompts feature flag.
- [ ] Overwrite prompts based on the date.

## To test

- Launch the Jetpack app.
- Log in to a WP.com account.
- 🔎 Verify that the blogging prompt card appears on the home dashboard.
- Tap "View all responses".
- 🔎 Verify that the Reader stream shows the posts related to the prompt.
- Go back, and tap "Answer prompt".
- 🔎 Verify that the draft post is opened and shows the prompt content in quotes.
- Go back, tap the ellipsis menu, and select "View more prompts".
- 🔎 Verify that the lists are loaded correctly, and no duplicated dates are shown.

#### Prompt scheduling

- Install the app store version of Jetpack.
- Schedule a prompt.
- Update the Jetpack with this version and launch the app.
- Wait for the reminder to be fired.
- 🔎 Verify that the reminder shows the contents of the new v3 prompts.

## Regression Notes
1. Potential unintended areas of impact
This may be affecting scheduled prompts.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

3. What automated tests I added (or what prevented me from doing so)
Updated and ran the unit tests on both WordPress-iOS and WordPressKit-iOS.

PR submission checklist:


UI Changes testing checklist:
N/A.
